plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java

```diff
@@ -593,7 +597,7 @@ public void indexStatsCollectorJob(boolean enabled) {
                 .source(jobParameter.toXContent(JsonXContent.contentBuilder(), null))
                 .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
 
-            indexJob(indexRequest, MLJobType.STATS_COLLECTOR, () -> {});
+            indexJob(indexRequest, MLJobType.STATS_COLLECTOR, () -> this.statsCollectorJobStarted = enabled);
```

Comment:
indexJob(indexRequest, MLJobType.STATS_COLLECTOR, () -> {
            this.statsCollectorJobStarted = enabled;
            log.debug("Stats collector job {} successfully", enabled ? "started" : "stopped");
        });